package/samplers/grey_wolf_optimization/grey_wolf_optimization.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import annotations
+
+from typing import Any
+
+import numpy as np
+import optuna
+from optuna.distributions import BaseDistribution
+from optuna.samplers import RandomSampler
+import optunahub
+
+
+class GreyWolfOptimizationSampler(optunahub.load_module("samplers/simple").SimpleBaseSampler):  # type: ignore
+    def __init__(
+        self,
+        search_space: dict[str, BaseDistribution] | None = None,
+        population_size: int = 10,
+        max_iter: int = 40,
+        num_leaders: int = 3,
+        seed: int = 0,
+    ) -> None:
+        self.search_space = search_space
```

Comment:
We might want to call `SimpleBaseSampler.__init__`. We can work on it in a follow-up PR.
```suggestion
    ) -> None:
        super().__init__(search_space, seed)
        self.search_space = search_space
```